java/src/jmri/StringIOManager.java

```diff
@@ -1,10 +1,54 @@
 package jmri;
 
+import javax.annotation.Nonnull;
+import javax.annotation.CheckForNull;
+
 /**
  * Interface for obtaining StringIOs.
  * 
  * @author Daniel Bergqvist Copyright (C) 2020
  */
-public interface StringIOManager extends Manager<StringIO> {
+public interface StringIOManager extends ProvidingManager<StringIO>, NameIncrementingManager {
+
+    @Nonnull
+    StringIO provide(@Nonnull String name) throws IllegalArgumentException;
+
+    @Nonnull
+    StringIO provideStringIO(@Nonnull String name) throws IllegalArgumentException;
+
+    /**
+     * Return a StringIO with the specified user or system name.
+     * Return StringIO by UserName else provide by SystemName.
+     * <p>
+     * Note that
+     * two calls with the same arguments will get the same instance; there is
+     * only one StringIO object representing a given physical turnout and
```

Comment:
> turnout is a bit misleading here.
Fixed

> But a question: If I call newStringIO("Daniel") and the StringIO LNDaniel exists, then the StringIO LNDaniel will be returned?

The search order is user name exactly, system name exactly, then in the default system with the prefix added. So if there's an LNDaniel in the LocoNet (default) system, it'll be found.  Otherwise, not.